composer/callbacks/image_monitor.py

```diff
@@ -0,0 +1,194 @@
+            logger.data_batch({key: table})
+
+    def _log_segmented_inputs(self, state: State, logger: Logger, key: str):
+        inputs = state.batch_get_item(key=self.input_key)
+        targets = state.batch_get_item(key=self.target_key)
+        outputs = state.outputs
+
+        if not isinstance(inputs, torch.Tensor):
+            raise NotImplementedError('Multiple input tensors not supported yet')
+        if not isinstance(targets, torch.Tensor):
+            raise NotImplementedError('Multiple target tensors not supported yet')
+        if not isinstance(outputs, torch.Tensor):
+            raise NotImplementedError('Multiple output tensors not supported yet')
+
+        table = _make_segmentation_images(inputs, targets, outputs, self.num_images, self.shift_targets)
+        logger.data_batch({key: table})
+
+    def before_forward(self, state: State, logger: Logger):
+        assert isinstance(self.interval, Time)
+        if self.mode.lower() == 'input' and state.timestamp.get(self.interval.unit).value % self.interval.value == 0:
+            self._log_inputs(state, logger, 'Images/Train')
+
+    def eval_before_forward(self, state: State, logger: Logger):
+        assert isinstance(self.interval, Time)
+        if self.mode.lower() == 'input' and state.eval_timestamp.get(TimeUnit.BATCH).value == 0:
+            self._log_inputs(state, logger, 'Images/Eval')
+
+    def before_loss(self, state: State, logger: Logger):
+        assert isinstance(self.interval, Time)
+        if self.mode.lower() == 'segmentation' and state.timestamp.get(
+                self.interval.unit).value % self.interval.value == 0:
+            self._log_segmented_inputs(state, logger, 'Images/Train')
+
+    def eval_after_forward(self, state: State, logger: Logger):
+        assert isinstance(self.interval, Time)
+        if self.mode.lower() == 'segmentation' and state.eval_timestamp.get(TimeUnit.BATCH).value == 0:
+            self._log_segmented_inputs(state, logger, 'Images/Eval')
+
+
+def _make_input_images(inputs: torch.Tensor, num_images: int):
+    import wandb
+    if inputs.shape[0] < num_images:
+        num_images = inputs.shape[0]
+    images = inputs[0:num_images].data.cpu().permute(0, 2, 3, 1).numpy()
+    table = wandb.Table(columns=['Image'])
+    for image in images:
+        img = wandb.Image(image)
+        table.add_data(img)
+    return table
+
+
+def _make_segmentation_images(inputs: torch.Tensor,
+                              targets: torch.Tensor,
+                              outputs: torch.Tensor,
+                              num_images: int,
+                              shift_targets: bool = False):
+    import wandb
+    if min([inputs.shape[0], targets.shape[0], outputs.shape[0]]) < num_images:
+        num_images = min([inputs.shape[0], targets.shape[0], outputs.shape[0]])
+    images = inputs[0:num_images].data.cpu().permute(0, 2, 3, 1).numpy()
+    targets = targets[0:num_images].data.cpu().numpy()
```

Comment:
Could targets be one-hot? Like for mixup? Would that work?

Reply:
You're right, one-hots weren't being handled properly before. Fixed it so things should work correctly now.